src/gz/bufread.rs

```diff
@@ -304,9 +388,61 @@ pub struct GzDecoder<R> {
     multi: bool,
 }
 
+#[derive(Debug)]
+pub enum GzHeaderParsingState {
+    GzHeaderParsingStart = 0,
+    GzHeaderParsingXlen = 1,
+    GzHeaderParsingExtra = 2,
+    GzHeaderParsingFilename = 3,
+    GzHeaderParsingComment = 4,
+    GzHeaderParsingCrc = 5,
+}
+
+#[derive(Debug)]
+pub struct GzHeaderPartial {
+    buf: Vec<u8>,
+
+    state: GzHeaderParsingState,
+    flg: u8,
+    os: u8,
+    xlen: u16,
+    mtime: u32,
+    extra: Option<Vec<u8>>,
+    filename: Vec<u8>,
+    comment: Vec<u8>,
```

Comment:
Could this perhaps internally store a `GzHeader` and then it just appends a `buf` and `state` field?

Reply:
Good idea, will try

Reply:
Ah, we need at least `flg` and `xlen` in `GzHeaderPartial`
So, should I still try to put a `GzHeader` in `GzHeaderPartial` ?

Reply:
I don't know without doing it myself, but I'm trying to ideally reduce duplication here instead of listing the fields twice in a number of places.

Reply:
Right, doing so